### PR TITLE
fix(ssh): include endpoint in model deduplication

### LINKED ISSUE #944: [Bug] Legacy SSH model library deduplicates custom models without base URL
## Describe the bug

The legacy SSH model-library path treats `(provider, model)` as the complete identity when adding a model. It ignores `baseUrl`, unlike the local and Dashboard model-library paths.

When two OpenAI-compatible endpoints expose the same model ID, adding the model from the second endpoint returns the first endpoint's existing row and never writes the second attachment. The user can then only select the model through the first provider/endpoint.

## Steps to reproduce

1. Use an SSH connection with the legacy chat transport.
2. Add `shared-model` under a custom provider at `https://a.example/v1`.
3. Add the same model ID under another custom provider at `https://b.example/v1`.
4. Inspect the remote `~/.hermes/models.json`.

The second add returns the row for endpoint A, and the library still contains only one attachment.

## Expected behavior

The remote library should contain both attachments because endpoint URL is part of a model attachment's identity:

- `custom / shared-model / https://a.example/v1`
- `custom / shared-model / https://b.example/v1`

Equivalent URL variants, such as a trailing slash or host casing difference, should remain idempotent.

## Root cause

`sshAddModel` currently searches only by `m.model === model && m.provider === provider`. The local `addModel` implementation and the Dashboard compatibility endpoint both include a normalized base URL in their deduplication key.

## Impact

Users with multiple custom providers that expose the same model ID cannot attach that model to the second provider in legacy SSH mode. Subsequent selection can resolve to the wrong endpoint and credentials.

## Environment

- Hermes Desktop `main` at `3f744975f818bbb40ed029e6b3022cd0c5ad7a24`
- Connection mode: SSH
- SSH chat transport: legacy

The failure is reproducible offline with a mocked SSH file transport.

### PR DESCRIPTION
**Pull request by Quine-rq**
## Summary

- Include normalized `baseUrl` in legacy SSH model-library deduplication.
- Preserve separate attachments when two custom endpoints expose the same model ID.
- Keep equivalent endpoint URL variants idempotent.
- Document the shared attachment-identity rule across local, Dashboard, and legacy SSH transports.

Fixes #944

## Why

Local `addModel` and the Dashboard compatibility endpoint identify a model attachment by `(provider, model, normalized base URL)`. `sshAddModel` only used `(provider, model)`, so the second of two custom endpoints serving the same model ID was silently discarded and the first endpoint's row was returned.

## Validation

- `npm test` — 198 files passed; 1,957 tests passed, 3 skipped.
- `npm run lint`
- `npm run typecheck`
- `npx prettier --check src/main/ssh-remote.ts tests/ssh-model-library.test.ts lat.md/provider-setup.md`
- `npx --yes lat.md check`

The regression suite covers both distinct endpoints and normalized URL variants using a mocked SSH read/write transport; it requires no live SSH host or credentials.

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation-only change

## Checklist

- [x] Code follows the repository style.
- [x] Full tests, lint, typecheck, formatting, and lat.md validation pass.
- [x] Tests cover the failure and the idempotent path.
- [x] No credentials or user data are included.
- [x] AI assistance was used; the change and tests were manually reviewed.

### CODE REVIEW

File: src/main/ssh-remote.ts [hunk 1]
```diff
@@ -3522,8 +3522,13 @@ export async function sshAddModel(
   baseUrl: string,
 ): Promise<SavedModel> {
   const models = await sshListModels(config);
+  const norm = (url: string): string =>
+    (url || "").trim().replace(/\/+$/, "").toLowerCase();
```
[the remark this answers is not in the document]
Comment (Quine-rq) [reply src/main/ssh-remote.ts]: Fixed in b05bc6d. Endpoint identity now normalizes only case-insensitive URL components (scheme, host, and default port) plus trailing path slashes, while preserving case-sensitive path, query, credential, and fragment components. The same shared normalization is used across local, Dashboard, Remote, and legacy SSH paths.

Added the requested TenantA/tenanta regression plus focused normalization coverage, including IPv6. Validation: 199 test files passed (1,962 tests passed, 3 skipped), with lint, both typechecks, Prettier, lat.md check, and git diff --check all passing. Please re-review this commit.

**Comment by Quine-rq** [comment]
Addressed the case-sensitive-path review in b05bc6d. Endpoint identity now normalizes only case-insensitive URL components (scheme, host, and default port) plus trailing path slashes, while preserving path, query, credential, and fragment case. The same shared normalization is used across local, Dashboard, Remote, and legacy SSH paths.

Added the requested TenantA/tenanta regression and focused normalization coverage, including IPv6. Validation: 199 test files passed (1,962 tests passed, 3 skipped); lint, both typechecks, Prettier, lat.md check, and git diff --check all pass. Please re-review the latest commit.

**Comment by Quine-rq** [comment]
Addressed the dot-segment canonicalization review in 9b9f3c6. The injected Dashboard Python normalizer now matches the desktop WHATWG path shortening for literal and percent-encoded single/double-dot segments, while preserving repeated slashes and non-dot encoded segments. Added a regression that executes the actual injected Python and compares 21 edge cases against the shared TypeScript normalizer.

Validation: 199 test files passed (1,963 tests passed, 3 skipped); lint, both typechecks, Prettier, lat.md check, and git diff --check all pass. Please re-review the latest commit.

**Review by fathah — COMMENTED** [review]
Thanks @Quine-rq for the contribution and for improving Hermes One.

Reviewed endpoint-aware model identity across local, SSH, remote, and injected Python paths. Greptile’s path-case and dot-segment findings are addressed. Added a correction and regression for Python dropping explicit port zero. Local validation: 28 focused tests passed; the combined changes also pass the full suite in #946. Merge this before #946.

Both TypeScript checks, full lint, and lat check passed locally. Greptile’s latest review of this head is 5/5. Source review is complete; merge is pending the fork’s GitHub CI approval and a successful CI run.

Reviewed commit: `1ad0129f64a15aa13e222659ead682f4263c26b4`.

**Review by fathah — APPROVED** [review]
GitHub CI now passes on the reviewed commit: dependency audit, both TypeScript checks, the full test suite, and lint. Greptile also reports a successful review. The source findings and local validation are detailed in my earlier review. Approved for merge.

Reviewed commit: `1ad0129f64a15aa13e222659ead682f4263c26b4`.
